nrf_rpc/include/nrf_rpc.h

```diff
@@ -0,0 +1,444 @@
+ *
+ * This macro should be used if an response was allocated, but it will not be
+ * send with @a NRF_RPC_RSP_SEND.
+ *
+ * @param _ctx    Context that was previously allocated to send the response.
+ * @param _packet Packet that was previously allocated to send the response.
+ */
+#define NRF_RPC_RSP_DISCARD(_ctx, _packet)				       \
+	nrf_rpc_tr_free_tx_buf((_ctx), (_packet))
+
+
+/* Internal functions used by the macros only. */
+struct nrf_rpc_tr_remote_ep *_nrf_rpc_cmd_prep(
+	const struct nrf_rpc_group *group);
+void _nrf_rpc_cmd_alloc_error(struct nrf_rpc_tr_remote_ep *tr_remote_ep);
+void _nrf_rpc_cmd_unprep(void);
+struct nrf_rpc_tr_remote_ep *_nrf_rpc_evt_prep(
+	const struct nrf_rpc_group *group);
+void _nrf_rpc_evt_alloc_error(struct nrf_rpc_tr_remote_ep *tr_remote_ep);
+void _nrf_rpc_evt_unprep(struct nrf_rpc_tr_remote_ep *tr_remote_ep);
+struct nrf_rpc_tr_remote_ep *_nrf_rpc_rsp_prep();
+
+
+/** @brief Initialize the nRF RPC
+ *
+ * @returns         NRF_RPC_SUCCESS or error code from enum nrf_rpc_error_code.
+ */
+int nrf_rpc_init(void);
+
+
+/** @brief Send a command.
+ *
+ * @param remote_ep    Destination endpoint allocated by @a NRF_RPC_CMD_ALLOC.
+ * @param cmd          Command id.
+ * @param packet       Packet allocated by @a NRF_RPC_CMD_ALLOC and filled with
+ *                     encoded data.
+ * @param len          Length of the packet. Can be smaller than allocated.
+ * @param handler      Callback that handles the response. In case of error
+ *                     it is undefined if the handler will be called.
+ * @param handler_data Opaque pointer that will be passed to @a handler.
+ * @returns            NRF_RPC_SUCCESS or error code from
+ *                     enum nrf_rpc_error_code.
+ */
+int nrf_rpc_cmd_send(nrf_rpc_alloc_ctx ctx, uint8_t cmd, uint8_t *packet,
+		     size_t len, nrf_rpc_handler_t handler, void *handler_data);
+
+
+/** @brief Send a command and get response directly.
+ *
+ * After successful return caller is resposible for decoding content of the
+ * response packet an call @a nrf_rpc_decoding_done just after that. After
+ * calling @a nrf_rpc_decoding_done @a rsp_packet is no longer valid.
+ *
+ * Depending on transport layer implementation this function may be slightly
+ * slower than @a nrf_rpc_cmd_send, because additional thread context switching
+ * may happen.
+ *
+ * @param remote_ep       Destination endpoint allocated @a NRF_RPC_CMD_ALLOC.
+ * @param cmd             Command id.
+ * @param packet          Packet allocated by @a NRF_RPC_CMD_ALLOC and filled
+ *                        with encoded data.
+ * @param len             Length of the packet. Can be smaller than allocated.
+ * @param[out] rsp_packet Response packet.
+ * @param[out] rsp_len    Response packet length.
+ * @returns               NRF_RPC_SUCCESS or error code from
+ *                        enum nrf_rpc_error_code.
+ */
+int nrf_rpc_cmd_rsp_send(nrf_rpc_alloc_ctx ctx, uint8_t cmd, uint8_t *packet,
```

Comment:
Am I understand you correctly? You want to rename:
`nrf_rpc_cmd_send` -> `nrf_rpc_cmd_send_cb`
`nrf_rpc_cmd_rsp_send` -> `nrf_rpc_cmd_send`

Reply:
yes, just a suggestion. nrfxlib is not part of Zephyr, but consistent naming approach would help when using this library in Zephyr context. Then there is another:

```nrf_rpc_cbor_cmd_send_noerr``` -> ```nrf_rpc_cmd_send_cb_no_err```

Reply:
I can rename it this way, but I am thinking if this will introduce some confusion. Those functions are not async / sync variants. All three functions are sync. Calling function `_cb` may suggest that this is async function, but actually is sync (callback is called when the function is running).

Reply:
okay, then let's leave it as it is. Too bad that C does not support function overloading